app/services/utils/current_work_stream_calculator.rb

```diff
@@ -0,0 +1,25 @@
+module Utils
+  class CurrentWorkStreamCalculator
+    attr_reader :work_stream_param, :session_work_stream, :user_competencies
+
+    def initialize(work_stream_param:, session_work_stream:, user_competencies:)
+      @work_stream_param = work_stream_param
+      @session_work_stream = session_work_stream
+      @user_competencies = user_competencies
+    end
+
+    def current_work_stream
+      if work_stream_param.present?
+        Types::WorkStreamType[work_stream_param]
+      elsif session_work_stream.present?
+        session_work_stream
+      elsif user_competencies.present?
```

Comment:
@willmcb the bug you found was because competencies are event sourced and when competencies are set to none having been previously set, it is an empty array rather than a nil value. This present check handles both nil and empty arrays. I've also added a spec to calculator spec